### ISSUE TITLE
[Enhancement] Using lower resolutions with fullscreen borderless only games.

### ISSUE THREAD

**Issue by Aetopia** [SpecialKO/SpecialK#17]
Special K allows the user to play the game at any resolution they want.        
Even FSB only games are supported except when the game is closed or if the user switches windows, the display resolution isn't restored to native.

But actually a simple fix can be implemented to mitigate this.                      

If possible check if the game window has focus or not periodically, in case if the game window has focus switch the resolution else switch back to native.

**Comment by Aemony** [comment]
A note about this is that this would basically replicate FSE's behavior on Alt+Tab... Though it can be argued whether that's a positive or a negative due to the latency of forcing a display mode switch every time the user is switching to and from the game.

**Comment by Aetopia** [comment]
> A note about this is that this would basically replicate FSE's behavior on Alt+Tab... Though it can be argued whether that's a positive or a negative due to the latency of forcing a display mode switch every time the user is switching to and from the game.
> 
> In general, I'd probably say the easier option would be to:
> 
>     * Improve SK's actual use of FSE ?
> 
>     * Add support for "DirectX 12's FSE" if possible... Basically is that something that can be used by Special K to offload all of this to the OS itself so SK itself doesn't have to bother with it?


I made my own little utility a while back > https://github.com/Aetopia/Resolution-Enforcer

Which attempts to replicate FSE's Alt-Tab behavior.

There seems to be some form of Display Switching Latency when the user attempts to switch to a low resolution of lets say 1280x720 depending if the game is intensive or not. 1600x900 doesn't have that much delay in that case.

**Comment by Aemony** [comment]
I believe this feature, or something similar to it, has been implemented as of half a year or more ago?

**Comment by Aetopia** [comment]
> I believe this feature, or something similar to it, has been implemented as of half a year or more ago?

Yes, the feature exists, my only problem is that it doesn't "revert/restore" the resolution upon window deactivation.
